### ISSUE TITLE
Loss function: additional robustness?

### ISSUE THREAD

**Issue by blukaz** [yuxumin/PoinTr#19]
Hi @yuxumin, I've been wondering if you tried to combine both the CD_l1 & CD_l2 loss function during training? Maybe it would yield better results?

**Comment by yuxumin** [comment]
Hi, @blukaz . We haven't try it.

**Comment by blukaz** [comment]
I'll try it out and let you know.
Btw, have you tried EMD? Do you know any good and valid implementation for that?

**Comment by yuxumin** [comment]
@blukaz Thanks a lot.  We haven't tried to optimize emd loss along with CD loss during the training, but i think it worth a try. As for implementation for emd, maybe you can try [it](https://github.com/Colin97/MSN-Point-Cloud-Completion/tree/master/emd)  (But I can't guarantee the correstness of this implementation)

**Comment by blukaz** [comment]
@ thank you for the link! Keep you updated in the coming days.

**Comment by blukaz** [comment]
@yuxumin do you have any recommendation on how to fix clustering of the coarse prediction? I have the problem that the dense loss is a magnitude lower than the coarse. Did you encounter similar problems?

**Comment by yuxumin** [comment]
You mean that the CD-loss for dense cloud pairs are much less than those for coarse pairs?  Yes, i notice it too.  I find some works adopt the weighted sum of these two losses. For example, `loss = fine_loss + \alpha * coarse_loss` and decrease the value of \alpha during the training. But i am not sure whether it will help.

**Comment by blukaz** [comment]
Yes, exactly that. Ok, I see. I'll try that. Thank you for your quick response!

**Comment by m-kafiyan** [comment]
@blukaz do you have any updates regarding adding weight to the loss? I face similar problem
